src/client/Microsoft.Identity.Client/Cache/Items/MsalAccessTokenCacheItem.cs

```diff
@@ -13,11 +13,6 @@ namespace Microsoft.Identity.Client.Cache.Items
 {
     internal class MsalAccessTokenCacheItem : MsalCredentialCacheItemBase
     {
-        internal MsalAccessTokenCacheItem()

```

Comment:
This is a pretty long PR just because we used SortedSet in a lot of places, but the main improvements are in ScopeHelper.cs and associated tests ScopeHelperTests.cs